sycl/source/detail/scheduler/commands.hpp

```diff
@@ -633,7 +633,7 @@ pi_int32 enqueueImpKernel(
     const detail::EventImplPtr &Event,
     const std::function<void *(Requirement *Req)> &getMemAllocationFunc,
     sycl::detail::pi::PiKernelCacheConfig KernelCacheConfig,
-    bool KernelIsCooperative);
+    bool KernelIsCooperative, const bool KernelUsesClusterLaunch);
```

Comment:
If we are going to keep adding launch configurations (cooperative, cluster, etc.) we should consider packing them into bitmask or similar structure, just to avoid having to add a new bool in all these functions.

Reply:
> we should consider packing them into bitmask or similar structure, just to avoid having to add a new bool in all these functions.

Yes that definitely would need to be done, otherwise the function arguments will keep on expanding, but 
I believe this can be a part of a bigger refactor as we need this feature ASAP on the nightlies. 

Though, personally I would say that a  bitmask might not be the best for readability one would have to remember which bit represents which property. I would personally be of the opinion that we could pass the property list itself to this function (forcing the definition of this function to come to source file, a considerable refactor) and then something like - 
```cpp
if constexpr(std::tuple_size(properties) ! = 0) {
    ....
} else {
   ....
}
```

But again, I am of the opinion this can be a part of a later PR, mostly because of the urgency associated with this feature and also because we can then come up with  a proper solution as well, and then we can bundle up KernelCacheConfig, KernelIsCooperative and KernelUsesClusterLaunch and future properties together.

Reply:
> If we are going to keep adding launch configurations (cooperative, cluster, etc.) we should consider packing them into bitmask or similar structure, just to avoid having to add a new bool in all these functions.

Yes I think it is inevitable that more launch properties are added later on, and that there should be a long term solution such as the one you suggest. Whilst `cooperative` is the only one currently implemented that is supported across cuda/hip/l0, most of the ones that we could additionally implement for cuda are inspired by very backend agnostic gpgpu problems.
As @AD2605 mentioned we focused on simplicity in this time-pressured PR, and I agree that a agreed upon longer term solution should be made in a follow up PR.

There is for example one which allows the partial overlap of kernels (setup and teardown but not computation) within an in-order queue (stream), that is particularly targeted towards small kernels within applications comprising many short kernels (that is often the case in applications using graphs).
A less novel launch property is a new convenient way of setting the required local (cuda shared) memory size prior to kernel launch.

Reply:
That's alright. It seems like something we can address later, given it is not crossing the library boundary.